R/zchunk_L254.transportation_USA.R

```diff
@@ -88,100 +113,461 @@ module_gcam.usa_L254.transportation_USA_DISABLED <- function(command, ...) {
+    # Write all possible pass-through technologies to all regions
+
+    # Start with the table of output by tranTechnology
+    L254.StubTranTechOutput_USA %>%
+      group_by(region, year, supplysector) %>%
+      summarise(output_agg = sum(output)) %>%
+      ungroup() ->
+      L254.StubTranTechCalInput_passthru_USA_agg
+
+    A54.globaltech_passthru %>%
+      repeat_add_columns(tibble(year = BASE_YEARS)) %>%
+      repeat_add_columns(tibble(region = gcamusa.STATES)) %>%
+      select(region, supplysector, tranSubsector, stub.technology = technology, year, minicam.energy.input) %>%
+      # Subset only the passthrough technologies that are applicable in each region
+      semi_join(L254.StubTranTech_passthru_USA,
+                by = c("region", "supplysector", "tranSubsector", "stub.technology")) %>%
+      left_join(L254.StubTranTechCalInput_passthru_USA_agg,
+                by = c("region", "year", "minicam.energy.input" = "supplysector")) %>%
+      replace_na(list(output_agg = 0)) %>%
+      arrange(desc(minicam.energy.input)) %>%
+      group_by(region, year) %>%
+      mutate(output_cum = cumsum(output_agg)) %>%
+      ungroup() ->
+      L254.StubTranTechCalInput_passthru_USA_cum
+
+    LIST_supplysector <- unique(L254.StubTranTechCalInput_passthru_USA_cum$supplysector)
+
+    L254.StubTranTechCalInput_passthru_USA_cum %>%
+      mutate(calibrated.value = if_else(minicam.energy.input %in% LIST_supplysector,
+                                        output_cum, output_agg)) %>%
+      mutate(share.weight.year = year,
+             subs.share.weight = if_else(calibrated.value > 0, 1, 0),
+             tech.share.weight = if_else(calibrated.value > 0, 1, 0)) %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["StubTranTechCalInput"]]))) ->
+      L254.StubTranTechCalInput_passthru_USA
+
+    # L254.BaseService_trn_USA: base-year service output of transportation final demand
+    L254.StubTranTechOutput_USA %>%
+      select(one_of(c(LEVEL2_DATA_NAMES[["StubTranTech"]])), year, base.service = output) %>%
+      bind_rows(L254.StubTranTechProd_nonmotor_USA %>%
+                  select(one_of(c(LEVEL2_DATA_NAMES[["StubTranTech"]])), year, base.service = calOutputValue)) %>%
+      left_join_error_no_match(select(A54.sector, supplysector, energy.final.demand), by = "supplysector") %>%
+      group_by(region, energy.final.demand, year) %>%
+      summarise(base.service = sum(base.service)) %>%
+      ungroup ->
+      L254.BaseService_trn_USA
+
     # ===================================================
 
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
+    L254.DeleteSupplysector_USAtrn %>%
       add_title("descriptive title of data") %>%
```

Comment:
Remember to fill in title, units, etc. for outputs

Reply:
Hi Ben, I haven't started working on any documentation or the tests. This is for you to look at the issue of not allowing to overwrite title for the outputs.  It does not generate any output now. Sorry I didn't make it clear!

Reply:
Oops, sorry! My mistake–you were clear. Hang on, will go look again.